drift_dev/pubspec.yaml

```diff
@@ -6,51 +6,51 @@ homepage: https://drift.simonbinder.eu/
 issue_tracker: https://github.com/simolus3/drift/issues
 
 environment:
-  sdk: '>=2.14.0 <3.0.0'
+  sdk: '>=2.17.0 <3.0.0'
```

Comment:
> should I upgrade the drift_dev version to 2.1.1?

Actually it should be `2.2.0-dev`, there have been some new features in the meantime. I will strip the `-dev` before releasing, but you can also leave the version as it is.

In general, I don't like raising the minimum dependency constraints if we don't need it. It's fine for the language version I guess, but do we really need a new feature that has been added in `charcode` version `1.3.1`? If we support the lower version, our pubspec should indicate that as well. We want to test against the latest versions we claim to support, which is why the CI runs `pub upgrade`. There's no need to _only_ support the latest version by raising the constraint in the pubspec.

So, can we keep the constraints of the other packages? Given that we don't seem to be affected by any breaking changes in `analyzer`, I suggest doing this:

```dart
  analyzer: '>=4.5.0 < 6.0.0'
  analyzer_plugin: '^0.11.0'
```

If the lower bound has been wrong before and needed to be upgraded - that's fine! But I doubt we need to raise it to the latest version for every single dependency.